src/backend/distributed/commands/extension.c

```diff
@@ -491,6 +495,137 @@ PreprocessAlterExtensionUpdateStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PostprocessAlterExtensionCitusUpdateStmt is called after ALTER EXTENSION
+ * citus UPDATE command is executed by standard utility hook.
+ *
+ * Actually, we do not need such a post process function for ALTER EXTENSION
+ * UPDATE commands unless the extension is Citus itself. This is because we
+ * need to mark existing objects that are not included in distributed object
+ * infrastructure in older versions of Citus (but now should be) as distributed
+ * if we really updated Citus to the available version successfully via standard
+ * utility hook.
+ */
+List *
+PostprocessAlterExtensionCitusUpdateStmt(Node *node, bool
+										 citusCanBeUpdatedToAvailableVersion)
+{
+	if (!CitusHasBeenLoaded())
+	{
+		return NIL;
+	}
+
+	/*
+	 * We should not postprocess this command in workers as they do not keep track
+	 * of citus.pg_dist_object.
+	 */
+	if (!IsCoordinator())
+	{
+		return NIL;
+	}
+
```

Comment:
maybe continue with the early returns and comments, e.g.:
```c
/*
 * Prior to the ALTER EXTENSION citus UPDATE, Citus was already at the latest
 * available version, meaning the ALTER EXTENSION had no effect.
 */
if (!citusCanBeUpdatedToAvailableVersion)`
```
I like that pattern because it's easy to add comments.

Reply:
I couldn't get it quite, should I move this check to utility_hook, before calling Postprocess function ?

Reply:
This variable is currently checked in the `if (citusCanBeUpdatedToAvailableVersion && ... && ..)` below, but might as well return here, or at the top of this function.

Reply:
now got it